src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -252,9 +252,19 @@ CHIP_ERROR CodegenDataModelProvider::EventInfo(const ConcreteEventPath & path, D
     {
         return cluster->EventInfo(path, eventInfo);
     }
-
-    eventInfo.readPrivilege = RequiredPrivilege::ForReadEvent(path);
-    return CHIP_NO_ERROR;
+    const EmberAfCluster * cluster = emberAfFindServerCluster(path.mEndpointId, path.mClusterId);
```

Comment:
Why do we need this check? I imagine:

- if doing wildcard expansion, then the path will be valid already
- if not doing wildcard expansion, the caller can validate the path if needed

Reply:
This function will not accept a wildcard event path as the input path is a concrete path.

And for the concrete event path, the codegen data model might not support that event.

In zap code, there are event lists in the generated cluster structures https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/zap/tests/outputs/all-clusters-app/app-templates/endpoint_config.h#L3607-L3608, if the event is not supported, the list will not include the event id of that event.

So I add this check in EventInfo function.

Reply:
> In zap code, there are event lists in the generated cluster structures

That data is generally wrong.  We don't do anything with it right now, because of that, and if we start it's likely to break existing applications.

What we should probably do is stop generating that stuff (because it's just wasted codesize) and fix the ZAP UI to not even try to expose whether events are supported.

If an event is being generated, it's supported.